tests/qa-activity-report/generate-qa-activity-report.go

```diff
@@ -0,0 +1,201 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	monthStr := os.Getenv("MONTH")
+	yearStr := os.Getenv("YEAR")
+	daysStr := os.Getenv("DAYS")
+
+	var since, until, period string
+
+	if monthStr != "" && yearStr != "" {
+		month, err := strconv.Atoi(monthStr)
+		if err != nil {
+			fmt.Println("Invalid month:", err)
+			return
+		}
+		year, err := strconv.Atoi(yearStr)
+		if err != nil {
+			fmt.Println("Invalid year:", err)
+			return
+		}
+		startDate := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
+		endDate := startDate.AddDate(0, 1, -1)
+		since = startDate.Format("2006-01-02")
+		until = endDate.Format("2006-01-02")
+		period = fmt.Sprintf("%02d_%04d", month, year)
+	} else if daysStr != "" {
+		days, err := strconv.Atoi(daysStr)
+		if err != nil {
+			fmt.Println("Invalid number of days:", err)
+			return
+		}
+		until = time.Now().Format("2006-01-02")
+		since = time.Now().AddDate(0, 0, -days).Format("2006-01-02")
+		period = fmt.Sprintf("Last_%d_days", days)
+	} else {
+		fmt.Println("Please provide either MONTH and YEAR or DAYS.")
+		return
+	}
+
+	logs, err := getGitLog(since, until)
+	if err != nil {
+		fmt.Println("Error:", err)
+		return
+	}
+
+	csvRows := make([][]string, 0, 1000)
+
+	for _, logLine := range logs {
+		logParts := strings.Split(logLine, " ")
+		if len(logParts) < 2 {
+			continue
+		}
+
+		commit := logParts[0]
+		date := logParts[1]
+
+		diffLines, err := getGitDiff(commit)
+		if err != nil {
+			fmt.Println("Error:", err)
+			return
+		}
+
+		addedTests, changedTests, deletedTests := 0, 0, 0
+		var inExamples bool
+		var inScenarios bool
+
+		for i := 0; i < len(diffLines); i++ {
+			line := diffLines[i]
+
+			if strings.HasPrefix(line, "+") && !strings.HasPrefix(line, "+++") {
+				if strings.Contains(line, "Scenario:") {
+					addedTests++
+				} else if strings.Contains(line, "Scenario Outline:") {
+					// count the number of lines in the examples
+					inExamples = false
+					for j := i + 1; j < len(diffLines); j++ {
+						exampleLine := diffLines[j]
+						if strings.HasPrefix(exampleLine, "+") && strings.Contains(exampleLine, "Examples:") {
+							inExamples = true
+							continue
+						} else if inExamples {
+							trimmedLine := strings.TrimSpace(exampleLine)
+							if strings.HasPrefix(trimmedLine, "+") && strings.Contains(trimmedLine, "|") {
+								// Count a string if it starts with "+" and contains "|"
+								addedTests++
+							} else if strings.TrimSpace(exampleLine) == "" || !strings.HasPrefix(trimmedLine, "+") || !strings.HasPrefix(trimmedLine, "|") {
+								// Abort counting when a row that does not belong to the table is encountered
+								break
+							}
+						}
+					}
+					// We have one line | resource | which is not a test line. So we deleted one line from addedTest
+					// Examples:
+					// 	| resource      |
+					// 	| testfile.txt  |
+					// 	| FolderToShare |
+					if inExamples {
+						addedTests--
+					}
```

Comment:
Can we move this in a separate function that returns the number of added test in this scenario outline? That would make it much easier to read.